### DISCUSSION TITLE
What is the latest way to configure the helm and kubernetes providers?

### DISCUSSION THREAD

**Discussion by yorinasub17** [gruntwork-io/knowledge-base#352] [Help] [answered]
We had been following gruntworks recommendation for how to configure helm provider to ensure the token refresh:

```
provider "helm" {
  kubernetes {
    host                   = "${eks_cluster_endpoint}"
    cluster_ca_certificate = base64decode("${eks_certificate_authority}")
    exec {
      api_version = "client.authentication.k8s.io/v1alpha1"
      command     = "aws"
      args        = ["eks", "get-token", "--cluster-name", "${eks_cluster_name}"]
    }
  }
}
```

However more recently we are seeing a lot of apply errors with:

```
Error: Kubernetes cluster unreachable: Get "https://xxxxxxx.eks.amazonaws.com/version?timeout=32s": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
```

Has something changed, should we be using different helm provider configuration?

**Comment by yorinasub17** [comment, marked as answer]
We still use this method both internally at Gruntwork and in our Reference Architecture. AFAIK, we haven't run into any issues using this configuration.

The error message `request canceled while waiting for connection` suggests this is more likely a network error reaching the EKS kubernetes endpoint. Do you use private endpoints (where the k8s API is only accessible from within the VPC)? It's possible that there is some issue with the VPN connection: we've had issues in the past where the MTU settings from OpenVPN led to dropped packets, causing all sorts of network errors reaching private endpoints.

**Reply by lorelei-rupp-imprivata** [reply]
This actually was super helpful.. I didn't even think about the EC2 instances we run our agents that service the pipelines that are running the terragrunt applies. I checked them out and they were resource starved. I rolled out bigger instances and the issue appears to have gone away.
